### ISSUE TITLE
Disable undo menu item when there is no more history available

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/vscode#55440]
From #55389

**Feature request**
Disable the undo menu item if there is no undo history available. 

(same for redo if there is no more forward history)

**Comment by bpasero** [comment]
Shouldn't the editor provide this a as a context key @alexandrudima ?

**Comment by alexandrudima** [comment]
This is blocked by #55453, currently the global context is evaluated when computing the enablement/disablement of extensions.

**Comment by sbatten** [comment]
we would need an `activeEditorCanUndo` key much like `activeEditorCanSave`

**Comment by kanlukasz** [comment]
Definitely a very, very important function.
We need to know if there is something to undo or repeat. Sometimes it's **very important**

### Examples from other editors:

#### N++ - no redo
![obraz](https://user-images.githubusercontent.com/30685349/70305123-dffba180-1803-11ea-9d50-d85ab50a7850.png)

#### N++ - no undo
![obraz](https://user-images.githubusercontent.com/30685349/70305133-e9850980-1803-11ea-8b31-3741cc699a81.png)

#### ST3 - no undo
![obraz](https://user-images.githubusercontent.com/30685349/70305353-6fa15000-1804-11ea-970d-dc6747e5969e.png)

#### ST3 - no redo
![obraz](https://user-images.githubusercontent.com/30685349/70305380-8b0c5b00-1804-11ea-8978-cfb745070ac6.png)

I keep my fingers crossed for it

**Comment by kanlukasz** [comment]
Is that still blocked by #55453 ?
I ask because #55453 is closed

**Comment by sbatten** [comment]
@alexdima I know undo state has been modified recently, any changes that would help here?

**Comment by alexdima** [comment]
@sbatten Yes, the undo redo service can be queried based on the current focused editor and asked if there is anything to undo or to redo. But this still depends on focus, so we would need to have a global context key that updates every time focus is changed.

**Comment by psnet** [comment]
Any movement?.. 2023 is now. Sometimes this is must, and it still appears unresolved
